src/components/TextArea/index.tsx

```diff
@@ -0,0 +1,113 @@
+import React, {useEffect, useState} from 'react';
+import useStyles from './styles';
+import {
+  NativeSyntheticEvent,
+  TextInput as RNTextInput,
+  TextInputFocusEventData,
+  TextInputProps,
+  Text,
+  View,
+} from 'react-native';
+import {useTheme} from '../Theme';
+import ErrorMessage from '../ErrorMessage';
+import Icon from '../Icon';
+
+export interface IManualControlProps extends TextInputProps {
+  error?: boolean;
+  errorText?: string;
+  disabled?: boolean;
+  textLimit?: number;
+  name?: string;
+}
+
+const TextArea: React.FC<IManualControlProps> = props => {
+  const {
+    errorText,
+    disabled = false,
+    textLimit,
+    name = 'Label',
+    onChangeText,
+    onFocus,
+    onBlur,
+    value,
+    ...other
+  } = props;
+  const [focused, setFocused] = useState(false);
+  const styles = useStyles();
+  const {theme} = useTheme();
+
+  const onFocusWrapper = (e: NativeSyntheticEvent<TextInputFocusEventData>) => {
+    setFocused(true);
+    onFocus?.(e);
+  };
+  const onBlurWrapper = (e: NativeSyntheticEvent<TextInputFocusEventData>) => {
+    setFocused(false);
+    onBlur?.(e);
+  };
+
+  const onChangeTextWrapper = (text: string) => {
+    onChangeText?.(text);
+  };
+
+  return (
+    <View>
+      <View
+        style={[
+          styles.container,
+          focused && styles.focused,
+          !!errorText && styles.error,
+          disabled && styles.disabled,
+        ]}
+      >
+        <Text style={[styles.label, disabled && styles.textDisabled]}>
+          {name}
+        </Text>
+        <RNTextInput
+          maxLength={textLimit}
+          onFocus={onFocusWrapper}
+          onBlur={onBlurWrapper}
+          placeholderTextColor={theme.text.disabled}
+          placeholder="Type something..."
+          multiline
+          style={styles.textArea}
+          editable={!disabled}
+          onChangeText={onChangeTextWrapper}
+          value={value}
+          {...other}
+        />
+      </View>
+      <View
+        style={[
+          styles.bottomContainer,
+          (disabled || (!disabled && !errorText)) && styles.bottomContainerEnd,
+        ]}
+      >
+        {(errorText?.length ?? 0) > 0 && !disabled && (
+          <View style={styles.errorContainer}>
+            <Icon
+              width={20}
+              height={20}
+              name="error"
+              filled
+              color={theme.error.main}
+            />
+
+            {(errorText?.length ?? 0) > 0 && (
```

Comment:
I think it can be just `errorText && <ErrorMessage />`

Reply:
also this condition is not needed here as you have declared it at 85 already